test/legacy_test/test_sparse_pca_lowrank.py

```diff
@@ -0,0 +1,147 @@
+#   Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import random
+import re
+import unittest
+
+import numpy as np
+
+import paddle
+
+
+def get_cuda_version():
+    result = os.popen("nvcc --version").read()
+    regex = r'release (\S+),'
+    match = re.search(regex, result)
+    if match:
+        num = str(match.group(1))
+        integer, decimal = num.split('.')
+        return int(integer) * 1000 + int(float(decimal) * 10)
+    else:
+        return -1
+
+
+class TestSparsePcaLowrankAPI(unittest.TestCase):
+    def transpose(self, x):
+        shape = x.shape
+        perm = list(range(0, len(shape)))
+        perm = perm[:-2] + [perm[-1]] + [perm[-2]]
+        return paddle.transpose(x, perm)
+
+    def random_sparse_matrix(self, rows, columns, density=0.01, **kwargs):
+        dtype = kwargs.get('dtype', paddle.float64)
+
+        nonzero_elements = max(
+            min(rows, columns), int(rows * columns * density)
+        )
+
+        row_indices = [i % rows for i in range(nonzero_elements)]
+        column_indices = [i % columns for i in range(nonzero_elements)]
+        random.shuffle(column_indices)
+        indices = [row_indices, column_indices]
+        values = paddle.randn((nonzero_elements,), dtype=dtype)
+        values *= paddle.to_tensor(
+            [-float(i - j) ** 2 for i, j in zip(*indices)], dtype=dtype
+        ).exp()
+        indices_tensor = paddle.to_tensor(indices)
+        x = paddle.sparse.sparse_coo_tensor(
+            indices_tensor, values, (rows, columns)
+        )
```

Comment:
we also need to test the sparse_csr_tensor

Reply:
Please add a TODO to remind us to complement the test when sparse.sum with axis(-2) is implemented in the future.

Reply:
Done